pages/catalog/CatalogBookHandler.inc.php

```diff
@@ -110,6 +111,9 @@ function book($args, $request) {
 
 			// Expose variables to template
 			$templateMgr->assign('availableFiles', $filteredAvailableFiles);
+		} else {
+			error_log("payment manager not configured");
+			$templateMgr->assign('availableFiles', array());
```

Comment:
I think we need to deliver the files even if no payment manager is configured. @asmecher I'm a little unsure what the `if ($paymentManager->isConfigured())` is accomplishing here. Shouldn't `$filteredAvailableFiles` be passed to the template in either case?

Reply:
Historically, unconfigured payment tools shut off parts of the site confusingly, and we needed to remind users to configure it when they hit roadblocks. Instead, I'd rather we let the site behave usefully up until a configuration problem is encountered, then show an error. So yes, let's pass this to the template either way.

Reply:
Ok then, @paflov, could you just the `if` conditionals around this and pass the  `$filteredAvailableFiles` to the template in any case.